File: frontend/src/types/report.ts
import type { ClaimItem } from './claim'
import type { ComparisonMatrix } from './comparisonMatrix'
import type { CompetitorProfile } from './competitorProfile'
import type { DimensionQAScore, QAResult } from './qa'

export interface ReportParagraph {
  paragraph_id: string
  text: string
  claim_ids: number[]
  evidence_ids: number[]
}

export interface ReportSection {
  section_id: string
  title: string
  paragraphs: ReportParagraph[]
}

export interface ReportQualitySummary {
  dimension_body_qa_status?: string
  finalizer_grounding_status?: string
  final_status?: string
  final_score?: number | string | null
  dimension_avg?: number | string | null
  finalizer_score?: number | string | null
  dimension_weight?: number
  finalizer_weight?: number
  dimension_pass_threshold?: number
  finalizer_pass_threshold?: number
  high_risk_issue_count?: number
  blockers?: string[]
}

export interface ReportFinalizerQa {
  passed?: boolean
  score?: number | string | null
  issues?: Record<string, unknown>[]
  revision_round?: number
  pass_threshold?: number | string | null
}

export interface ReportItem {
  id: number
  task_id: number
  title: string
  content_markdown: string
  content_html?: string | null
  report_json?: {
    title?: string
    sections?: ReportSection[]
    quality_summary?: ReportQualitySummary
    dimension_qa_scores?: DimensionQAScore[]
    finalizer_qa?: ReportFinalizerQa
    [key: string]: unknown
  } | null
  created_at: string
  updated_at: string
}

export interface ReportEvidenceItem {
  id: number
  source_url: string
  source_title?: string | null
  source_type?: string | null
  chunk_text: string
  competitor_name?: string | null
  reliability_score?: string | number | null
}

export interface ReportResponse {
  report: ReportItem | null
  claims: ClaimItem[]
  evidence: ReportEvidenceItem[]
  qa_result?: QAResult | null
  profiles?: CompetitorProfile[]
  matrices?: ComparisonMatrix[]
}

File: backend/app/schemas/report.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict

from app.schemas.matrix import ComparisonMatrixItem
from app.schemas.profile import CompetitorProfileItem


class ReportItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    task_id: int
    title: str
    content_markdown: str
    content_html: str | None
    report_json: dict | None = None
    created_at: datetime
    updated_at: datetime


class ReportClaimItem(BaseModel):
    id: int
    task_id: int
    claim_text: str
    competitor_name: str | None = None
    claim_type: str | None = None
    dimension_key: str | None = None
    dimension_label: str | None = None
    confidence: Decimal | None = None
    risk_level: str | None = None
    evidence_ids: list[int] = []
    created_at: datetime


class ReportEvidenceItem(BaseModel):
    id: int
    source_url: str
    source_title: str | None = None
    source_type: str | None = None
    chunk_text: str
    competitor_name: str | None = None
    reliability_score: Decimal | None = None


class ReportResponse(BaseModel):
    report: ReportItem | None
    claims: list[ReportClaimItem] = []
    evidence: list[ReportEvidenceItem] = []
    qa_result: dict | None = None
    profiles: list[CompetitorProfileItem] = []
    matrices: list[ComparisonMatrixItem] = []
